PageObjectPattern/src/test/java/apiSerenityStep/SpotifyApiStep.java

```diff
@@ -1,114 +1,36 @@
-        return appToken;
-    }
-
-    @Step("sending get request")
-    public void sendGetRequest(String endpoint) {
-        responseGet = SerenityRest.given().auth().oauth2(accessToken).get(baseUrl + endpoint);
-
-    }
-
-    @Step("sending get request with app token")
-    public void sendGetRequestWithAppToken(String endpoint) {
-        responseGet = SerenityRest.given().auth().oauth2(appToken).get(baseUrl + endpoint);
-    }
-
-    @Step("check Get response status")
-    public int getResponseStatusCode() {
-        int responseCode = responseGet.then().extract().statusCode();
-        return responseCode;
-    }
-
-    @Step("check post response status code")
-    public int getPostResponseStatusCode() {
-
-        int responseCode = responsePost.then().extract().statusCode();
-        return responseCode;
-    }
-
-//to check one field in response
-    @Step("check valid data in Get response")
-    public void checkIfDataIsValidGet(String requiredField, String responseData) {
-
-        responseGet.then().assertThat().body(requiredField, equalTo(responseData));
-
-    }
-
-    @Step("sending POST request to create Playlist")
-    public void createPlayList(String endpoint, String playListName) {
-
-        responsePost = SerenityRest.given()
-                .body("{\n" + " \"name\":\"" + playListName + "\",\n" + " \"public\":\"true\"\n" + "}")
-                .header("Authorization", "Bearer " + accessToken).header("Accept", "string").post(baseUrl + endpoint);
-
-    }
-
-//to check one field in response
-    @Step("check Post response and return boolean result")
-    public boolean checkResponseAfterPost(String playListName) {
-
-        String fieldName = responsePost.getBody().jsonPath().getString("name");
-        boolean result;
-        result = (fieldName.equals(playListName));
-        return result;
-    }
-
-//to check many fields in response
-    @Step("check multiple fields in Post response")
-    public void checkFieldsInPostResponse(String playListName) {
-
-        responsePost.then().assertThat().body("name", equalTo(playListName)).body("owner.id", equalTo("czc11mg48dvwd16aeq0jofr3j")).body("owner.display_name",
-                equalTo("Yulia")).body("tracks.limit", equalTo(100));
-
-    }
-
-    @Step("check presense of required fields")
-    public boolean checkPresenseOfRequiredFields(String requiredField) {
-
-        ResponseBody body = responseGet.getBody();
-        String bodyStringValue = body.asString();
-        boolean result;
-        result = (bodyStringValue.contains(requiredField));
-        return result;
-
-    }
-
-    @Step("invalidate access token")
-    public void invalidateAccessToken() {
-        this.accessToken = "123";
+import java.util.HashMap;
+import java.util.Map;
+
+// This class is used to customize api calls
+public class SpotifyApiStep extends BaseStep {
+
+    // getting one field
+    @Step("getting one field from response")
```

Comment:
Well, the description might be improved as well as a method name, to show what exactly this method do.

Just a hint not only for this method and class, but for REST Assured in general - you can always return a response from your method and then later on in your Step Definitions class work with that response getting all needed information.